metrics/api/src/main/java/io/helidon/metrics/api/Timer.java

```diff
@@ -0,0 +1,178 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * Records timing information about large numbers of short-running events (e.g., HTTP requests).
+ */
+public interface Timer extends Meter, HistogramSupport {
+
+    /**
+     * Starts a timing sample using the default system clock.
+     *
+     * @return new sample
+     */
+    static Sample start() {
+        return MetricFactoryManager.INSTANCE.get().timerStart();
+    }
+
+    /**
+     * Starts a timing sample using the clock associated with the specified {@link io.helidon.metrics.api.MeterRegistry}.
+     *
+     * @param registry the meter registry whose clock is to be used for measuring the interval
+     * @return new sample with start time recorded
+     */
+    static Sample start(MeterRegistry registry) {
+        return MetricFactoryManager.INSTANCE.get().timerStart(registry);
+    }
+
+    /**
+     * Starts a timing sample using the specified clock.
+     *
+     * @param clock a clock to be used
+     * @return new sample with start time recorded
+     */
+    static Sample start(Clock clock) {
+        return MetricFactoryManager.INSTANCE.get().timerStart(clock);
+    }
+
+    /**
+     * Updates the statistics kept by the timer with the specified amount.
+     *
+     * @param amount duration of a single event being measured by this timer. If the amount is less than 0
+     *               the value will be dropped
+     * @param unit   time unit for the amount being recorded
+     */
+    void record(long amount, TimeUnit unit);
+
+    /**
+     * Updates the statistics kept by the timer with the specified amount.
+     *
+     * @param duration duration of a single event being measured by this timer
+     */
+    void record(Duration duration);
+
+    /**
+     * Executes the {@link java.util.function.Supplier} {@code f} and records the time spent invoking the function.
+     *
+     * @param f   function to be timed
+     * @param <T> return type of the {@link java.util.function.Supplier}
+     * @return return value from invoking the function {@code f}
+     */
+    <T> T record(Supplier<T> f);
+
+    /**
+     * Executes the {@link java.util.concurrent.Callable} {@code f} and records the time spent it, returning the
+     * callable's result.
+     *
+     * @param f   callable to be timed
+     * @param <T> return type of the {@link java.util.concurrent.Callable}
+     * @return return value from invoking the callable {@code f}
+     * @throws Exception exception escaping from the callable
+     */
+    <T> T recordCallable(Callable<T> f) throws Exception;
```

Comment:
This method should be also called `record`. It does not seem to conflict with other methods on this interface

Reply:
Agreed. (Micrometer calls this method `recordCallable` and I'd intended to change the name in our API but I missed this one.)

Reply:
Changed